Robot2018/src/org/usfirst/frc/team199/Robot2018/commands/RunScript.java

```diff
@@ -20,15 +22,30 @@ public RunScript(String scriptName) {
     			String cmdName = cmd[0];
     			String cmdArgs = cmd[1];
     			
+    			double[] point = new double[2];
+    			double rotation = 0;
+    			String parentheseless;
+    			String[] pointparts;
+    			if (AutoUtils.isDouble(cmdArgs)) {
+        			rotation = Double.valueOf(cmdArgs);
+        		} else if (AutoUtils.isPoint(cmdArgs)) {
+        			parentheseless = cmdArgs.substring(1, cmdArgs.length() - 1);
+        			pointparts = parentheseless.split(",");
+        			point[0] = Double.parseDouble(pointparts[0]);
+        		}
+    			
     			switch (cmdName) {
 	    			case "moveto":
 	    				addSequential(new AutoMoveTo(cmdArgs.split(" ")));
 	    				break;
 	    			case "turn":
-	    				addSequential(new AutoTurn(Double.parseDouble(cmdArgs)));
+	    				addSequential(new PIDTurn(Double.parseDouble(cmdArgs), Robot.dt, Robot.dt.getGyro()));
+	    				AutoUtils.setRot(rotation);
 	    				break;
 	    			case "move":
-	    				addSequential(new AutoMove(Double.parseDouble(cmdArgs)));
+	    				addSequential(new PIDMove(Double.parseDouble(cmdArgs), Robot.dt, new PIDSourceAverage(null, null)));
+	    				AutoUtils.setX(point[0]);
+	    				AutoUtils.setY(point[1]);
 	    				break;
```

Comment:
It looks like you are using the AutoUtils class to maintain what you think the current state of the robot is. If that is the case, perhaps this would be better maintained in a class with a more descriptive name.

More importantly, you appear to be setting the state based on what the script requested. Wouldn't the actual robot state as measured by its sensors be more accurate?